lib/plugins/aws/invokeLocal/index.js

```diff
@@ -177,8 +192,151 @@ class AwsInvokeLocal {
         this.options.context);
     }
 
-    throw new this.serverless.classes
-      .Error('You can only invoke Node.js, Python, Java & Ruby functions locally.');
+    return this.invokeLocalDocker();
+  }
+
+  checkDockerDaemonStatus() {
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['version']);
+      docker.on('exit', error => {
+        if (error) {
+          reject('Please start the Docker daemon to use the invoke local Docker integration.');
+        }
+        resolve();
+      });
+    });
+  }
+
+  checkDockerImage() {
+    const runtime = this.getRuntime();
+
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['images', '-q', `lambci/lambda:${runtime}`]);
+      let stdout = '';
+      docker.stdout.on('data', (buf) => { stdout += buf.toString(); });
+      docker.on('exit', error => (error ? reject(error) : resolve(Boolean(stdout.trim()))));
+    });
+  }
+
+  pullDockerImage() {
+    const runtime = this.getRuntime();
+
+    this.serverless.cli.log('Downloading base Docker image...');
+
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['pull', `lambci/lambda:${runtime}`]);
+      docker.on('exit', error => (error ? reject(error) : resolve()));
+    });
+  }
+
+  getLayerPaths() {
+    const layers = _.mapKeys(
+      this.serverless.service.layers,
+      (value, key) => this.provider.naming.getLambdaLayerLogicalId(key)
+    );
+
+    return BbPromise.all(
+      (this.options.functionObj.layers || this.serverless.service.provider.layers || [])
+        .map(layer => {
+          if (layer.Ref) {
+            return layers[layer.Ref].path;
+          }
+          const arnParts = layer.split(':');
+          const layerArn = arnParts.slice(0, -1).join(':');
+          const layerVersion = Number(arnParts.slice(-1)[0]);
+          const layerContentsPath = path.join(
+            '.serverless', 'layers', arnParts[6], arnParts[7]);
+          const layerContentsCachePath = path.join(
+            cachePath, 'layers', arnParts[6], arnParts[7]);
+          if (fs.existsSync(layerContentsPath)) {
+            return layerContentsPath;
+          }
+          let downloadPromise = BbPromise.resolve();
+          if (!fs.existsSync(layerContentsCachePath)) {
+            this.serverless.cli.log(`Downloading layer ${layer}...`);
+            mkdirp.sync(path.join(layerContentsCachePath));
+            downloadPromise = this.provider.request(
+              'Lambda', 'getLayerVersion', { LayerName: layerArn, VersionNumber: layerVersion })
+              .then(layerInfo => download(
+                layerInfo.Content.Location,
+                layerContentsPath,
```

Comment:
@dschep I think this should be downloading to `layerContentsCachePath` instead of `layerContentsPath`. Currently this code creates empty directory in `layerContentsCachePath` in first run and subsequent runs will have empty directory copied to `layerContentsPath`. Due to this handler fails to import layer in subsequent runs

Reply:
I've raised an issue for this -> https://github.com/serverless/serverless/issues/5990

Reply:
Good catch